packages/buidler-etherscan/src/solc/metadata.ts

```diff
@@ -46,9 +43,11 @@ export async function decodeSolcMetadata(bytecode: Buffer) {
   );
 
   const { decodeFirst } = await import("cbor");
-  // TODO: throw an error for decoding errors that are returned without being thrown
-  // E.g. cbor.decodeFirst(Buffer.from([])) === cbor.Decoder.NOT_FOUND
-  return decodeFirst(metadataPayload);
+  // The documentation for decodeFirst mentions the `required` option even though
+  // the type information is missing it.
+  // See http://hildjj.github.io/node-cbor/Decoder.html#.decodeFirst
+  const options: DecoderOptions = { required: true } as any;
```

Comment:
Instead of importing the `DecoderOptions` and having to do this, I'd rather do:

```
const options: any = { required: true }
return decodeFirst(metadataPayload, options)
```

(leaving the comment, of course)

Reply:
Hm, I see. I did it this way to let the typescript compiler choose the correct function. It selects the one with a callback function that returns `void` otherwise. To work around this, the return type could be annotated in the `decodeSolcMetadata` function too.

Reply:
Oh, I didn't know that, in that case it makes sense!